sympy/tensor/tensor.py

```diff
@@ -1815,7 +1815,7 @@ def tensor_heads(s, index_types, symmetry=None, comm=0):
     return thlist
 
 
-class TensExpr(Expr):
+class TensorExpr(Expr):
```

Comment:
I think that we should deprecate TensExpr etc.

Reply:
You're right, I added deprecation messages for old names.

Reply:
Great, but I think you should return a `TensorExpr` object to not break user's code, but only print the deprecation message.

So

```
    def __new__(cls, *args, **kw_args):
        return TensorExpr(*args, **kw_args)
```
(or what the exact syntax is).

Reply:
No problem, I added the constructor you proposed and the staticmethods to `TensExpr` and others.